Ladybird/AppKit/UI/Tab.swift

```diff
@@ -0,0 +1,307 @@
+        let tab = controller.window as! Tab
+        return tab.web_view.handle()
+    }
+
+    func loadURL(_ url: URL) {
+        self.tabController()?.loadURL(url)
+    }
+
+    func onLoadStart(_ url: URL, isRedirect: Bool) {
+        self.title = Ladybird.string_to_ns_string(url.serialize())
+        self.favicon = Tab.defaultFavicon
+        self.updateTabTitleAndFavicon()
+
+        self.tabController()?.onLoadStart(url, isRedirect: isRedirect)
+
+        if let inspectorController = self.inspectorController {
+            let inspector = inspectorController.window as! Inspector
+            inspector.reset()
+        }
+    }
+
+    func onLoadFinish(_ url: URL) {
+        if let inspectorController = self.inspectorController {
+            let inspector = inspectorController.window as! Inspector
+            inspector.inspect()
+        }
+    }
+
+    func onURLChange(_ url: URL) {
+        self.tabController()?.onURLChange(url)
+    }
+
+    func onBackNavigationEnabled(_ backEnabled: Bool, forwardNavigationEnabled: Bool) {
+        self.tabController()?.onBackNavigationEnabled(backEnabled, forwardNavigationEnabled: forwardNavigationEnabled)
+    }
+
+    func onTitleChange(_ title: ByteString) {
+        self.tabController()?.onTitleChange(title)
+        self.title = Ladybird.string_to_ns_string(title)
+        self.updateTabTitleAndFavicon()
+    }
+
+    func onFaviconChange(_ bitmap: Gfx.Bitmap) {
+        let faviconSize: CGFloat = 16
+
+        guard let png = try? Gfx.PNGWriter.encode(bitmap).get() else {
+            return
+        }
+
+        let data = NSData(bytes: png.data(), length: png.size())
+        let favicon = NSImage(data: data as Data)!
+        favicon.resizingMode = .stretch
+        favicon.size = NSSize(width: faviconSize, height: faviconSize)
+
+        self.favicon = favicon
+        self.updateTabTitleAndFavicon()
+    }
+
+    func onAudioPlayStateChange(_ playState: Web.HTML.AudioPlayState) {
+        guard let view = (self.web_view as? LadybirdWebView)?.view else { return }
+
+        switch playState {
+        case .paused:
+            if view.page_mute_state() == .unmuted {
+                self.tab?.accessoryView = nil
+            }
+        case .playing:
+            let button = NSButton(image: self.iconForPageMuteState(), target: self, action: #selector(togglePageMuteState(_)))
+            button.toolTip = self.toolTipForPageMuteState()
+            self.tab?.accessoryView = button
+        @unknown default:
+            fatalError("Unknown AudioPlayState")
+        }
+    }
+
+    func onFindInPageResult(currentMatchIndex: Int, totalMatchCount: Int?) {
+        self.searchPanel.onFindInPageResult(currentMatchIndex: currentMatchIndex, totalMatchCount: totalMatchCount)
+    }
+
+    override var isVisible: Bool {
+        didSet {
+            (self.web_view as? LadybirdWebView)?.handleVisibility(isVisible)
+            super.isVisible = isVisible
```

Comment:
Do we need to set super.isVisible in didSet? I thought from reading the docs that didSet was called after `set` and as such the value would ... already be set. if that makes sense.